com.unity.netcode.adapter.utp/Runtime/UnityTransport.cs

```diff
@@ -517,6 +518,27 @@ private void Update()
                 {
                     ;
                 }
+
+                
+                //TODO: Get the correct connection instead of m_ServerConnection
+                m_Driver.GetPipelineBuffers(m_LastSendPipeline, NetworkPipelineStageCollection.GetStageId(typeof(NetworkMetricsPipelineStage)),
```

Comment:
What is the reason for doing this only for the last pipeline sent to? There could have been sends on multiple pipelines between two `Update` calls.

Also for reference, every pipeline gets its own buffers. In `UnityTransport` there are three pipelines defined (`m_UnreliableFragmentedPipeline`, `m_UnreliableSequencedFragmentedPipeline`, and `m_ReliableSequencedFragmentedPipeline`). They'll each have their own network metrics context. So basically every connection will get three different metrics contexts.

Reply:
Oh, I was under the impression it was a single pipeline. That's good to know.